core/frontend/src/tools/EventController.ts

```diff
@@ -42,12 +55,16 @@ export class EventController {
   private addDomListeners(domType: string[], element: HTMLElement) {
     const vp = this.vp;
     const listener = (ev: Event) => {
-      ev.preventDefault();
+      // decoratorDiv is deprecated from the public API really, so this usage is fine
+      // eslint-disable-next-line deprecation/deprecation
+      if (!vp.decorationDiv.contains(ev.target as Element)) ev.preventDefault();
```

Comment:
Definitely not a fan of this...
Can you put this back the way it was and try adding ev.stopPropagation to your test where you add the "mouseDown" listener for your sampleHtmlDecoration instead?

Reply:
I can test that. But I could not get `stopPropagation` to allow running the default event in my tests with HTML decorations in the project linked in the description.

Reply:
the existing test definitely needs to be fixed so I may draft this PR to figure that out, still, see this existing html marker which fails. I use @bentley/imodel-react-hooks to have react render JSX into the `htmlElement` of the marker.
```tsx
const jsx = <div
          onMouseDown={(e) => e.stopPropagation()}
          onMouseUp={(e) => e.stopPropagation()}
          onMouseMove={(e) => e.stopPropagation()}
          onMouseOver={(e) => e.stopPropagation()}
          onMouseOut={(e) => e.stopPropagation()}
          onWheel={(e) => e.stopPropagation()}
          onTouchStart={(e) => e.stopPropagation()}
          onTouchEnd={(e) => e.stopPropagation()}
          onTouchCancel={(e) => e.stopPropagation()}
          onTouchMove={(e) => e.stopPropagation()}
          onMouseDownCapture={(e) => e.stopPropagation()}
          onMouseUpCapture={(e) => e.stopPropagation()}
          onMouseMoveCapture={(e) => e.stopPropagation()}
          onMouseOverCapture={(e) => e.stopPropagation()}
          onMouseOutCapture={(e) => e.stopPropagation()}
          onWheelCapture={(e) => e.stopPropagation()}
          onTouchStartCapture={(e) => e.stopPropagation()}
          onTouchEndCapture={(e) => e.stopPropagation()}
          onTouchCancelCapture={(e) => e.stopPropagation()}
          onTouchMoveCapture={(e) => e.stopPropagation()}
        >
          <div className={styles.markerContent}>
            <ContentComponent {...contentProps} />
          </div>
        </div>;
```
the full example [is here](https://github.com/MichaelBelousov/frontend-sample-showcase/blob/add-react-marker-sample/src/frontend-samples/react-marker-sample/ReactMarker.tsx).
In a different file [here](https://github.com/MichaelBelousov/frontend-sample-showcase/blob/c51875d4707d32afce94e31e21651e5330017cd5/src/frontend-samples/react-marker-sample/ReactMarkerUI.tsx#L60), if I uncomment the injection of the new EventController implementation in this PR, it works.

I don't remember which order they get hit in, but I don't believe you can get the default behavior without stopping the preventDefault, stopPropagation will also prevent the default behavior.

Reply:
stopPropagation does not prevent the default behavior, it just prevents the event from bubbling up, i.e. event only goes to your element.

Since your html element is a child of the viewport parentDiv, stopPropagation should be all you need so that you can click on links, etc. because without bubbling, preventDefault won't get called...that's my understanding at least.

I don't know anything about react unfortunately...might need to ask Dan East about that...

Would be nice to have an html decoration example in display-test-app so that this could be debugged more easily...

Reply:
React should do *mostly* the same thing. I will do some further investigation with non-react listeners, and figure out exactly why stopPropagation doesn't work in this case. I also expected it to work. When I have that done, I'll try to codify it the accompanying tests. I'll draft this PR until I have that done.

Reply:
using native non-react event handlers did fix this

Reply:
Interesting. The react events have to be built on the native browser events, so it would seem like you ought to be able to use them too. @DanEastBentley any thoughts/suggstions?

Reply:
React recently switched from a document-element listener emulation strategy to placing listeners on the actual elements in its emulation. I don't know the ramifications of those changes well, but I imagine since `ReactDOM` was rendering into the marker's HTML element separately something around that is why the events were not following the normal propagation rules.
Potentially the following could fix it but I haven't tried it yet and may just go with native events for what I'm doing:
```tsx
const jsx = <div onMouseDown={(e) => e.nativeEvent.stopImmediatePropagation()} />;
```

Reply:
The e.stopPropagation in a React event handler does not prevent the event from bubbling up to native handlers. We believe this has been fixed in React 17 and is one reason we want to upgrade to React 17.